docs/decisions/0003-custom-user-model.md

```diff
@@ -0,0 +1,34 @@
+# 0002 Use Custom User Model
+
+- Date: 2024-04-10
+- Author(s): [Chen Zhang][chen]
+- Status: `Active`
+
+## Decision
+
+We will use a custom User model instead of using Django built-in User model.
+
+## Context
+
+Django comes with a built-in User model which provides basic authentication and 
+authorisation features. This model includes fields like username, email, password,
+and essential methods for user management. However, there're some limitations of
+the default User model.
+
+The built-in User model is designed so users can login via username and password,
+and won’t be able to login with email and password. Also, we're unable to add new
+fields to the default user model.
+
+[The official Django documentation][document] highly recommends using a custom user model when
+start a Django project, because it’s hard to switch a user model out for another once
+the project is deployed and being used by real users,
+
+## Implications
+
+In some cases, users might lost access to their email address before they update their
+email address on the website or APP, and if the project uses email as primary user identifier
+it will be difficult for them to get their accounts back.
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** I would also add a paragraph here saying that we realise this the build user model we provide here might not be _perfect_ for all use cases, but by making a custom user model part of the template project we can insure it is as easy as possible to change and evolve that model later one. Something this is a lot more difficult should we stick with the built in one.‏

Reply:
Thanks, this makes lot of sense.